app/lib/features/pins/widgets/pin_item.dart

```diff
@@ -253,39 +134,30 @@ class _PinDescriptionWidgetConsumerState
       EasyLoading.showError(L10n.of(context).updateNameFailed(e));
     }
   }
-}
-
-// cancel submit buttons for create pin
-class _ActionButtonsWidget extends ConsumerWidget {
-  const _ActionButtonsWidget({
-    required this.pin,
-    required this.onSave,
-    required this.onCancel,
-  });
-
-  final ActerPin pin;
-  final void Function()? onSave;
-  final void Function()? onCancel;
 
-  @override
-  Widget build(BuildContext context, WidgetRef ref) {
-    final pinEdit = ref.watch(pinEditProvider(pin));
-    return Visibility(
-      visible: pinEdit.editMode,
-      child: Row(
-        mainAxisAlignment: MainAxisAlignment.end,
-        children: <Widget>[
-          OutlinedButton(
-            onPressed: onCancel,
-            child: Text(L10n.of(context).cancel),
-          ),
-          const SizedBox(width: 5),
-          ActerPrimaryActionButton(
-            key: PinItem.saveBtnKey,
-            onPressed: onSave,
-            child: Text(L10n.of(context).save),
-          ),
-        ],
+  // pin link widget
+  Widget _buildPinLink() {
+    final pinEdit = ref.watch(pinEditProvider(widget.pin));
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: TextFormField(
+        key: PinItem.linkFieldKey,
+        onTap: () async =>
+            !pinEdit.editMode ? await openLink(pinEdit.link, context) : null,
+        controller: _linkController,
+        readOnly: !pinEdit.editMode,
+        decoration: const InputDecoration(
```

Comment:
not quite sure I understand how editing the pin link works now ... but a single click just opens the link, right?

Reply:
Good catch, I missed it out completely. I need to manage it same as title and description. Like option from tap and menu item. I will do the require changes regarding the same.

Reply:
@gnunicorn 
I have did required fixes on pin page to have full functional support of our new UX pattern. Check below commit and reference video for the same.

Commit:
https://github.com/acterglobal/a3/pull/1860/commits/3f719ca8909203087c7bd3de2e24215892ae7aeb

Video:

https://github.com/acterglobal/a3/assets/51526480/a6ae6399-e71c-4a10-b6cc-cdee37bc5dd7